sdk-extensions/resources/src/main/java/io/opentelemetry/sdk/extension/resources/ContainerResource.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.extension.resources;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.sdk.resources.Resource;
+import io.opentelemetry.semconv.resource.attributes.ResourceAttributes;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.annotation.Nullable;
+
+/** Factory for {@link Resource} retrieving Container ID information. */
+public final class ContainerResource {
+
+  private static final Logger logger = Logger.getLogger(ContainerResource.class.getName());
+  private static final String UNIQUE_HOST_NAME_FILE_NAME = "/proc/self/cgroup";
+  private static final Pattern HEX_EXTRACTOR =
```

Comment:
This regex looks quite gnarly. Can we just iterate for the indexes of optional metadata?

```java
@Nullable
private static String getIdFromCgroupLine(String line) {
  // Start at the character after '-' if found or beginning of string (-1 + 1) if not.
  int startIndex = line.indexOf('-') + 1;
  int endIndex = line.lastIndexOf('.');
  if (endIndex < 0) {
    endIndex = line.length();
  }
  String id = line.substring(startIndex, endIndex);
  if (!OtelEncodingUtils.isValidBase16StringString(id)) {
    return null;
  }
  return id;
}
```

It's a few more lines of code then creating the pattern / matcher, but not that much and seems much more straightforward (an if/else instead of relying on `-1 + 1` is totally reasonable to make even more straightforward).

Reply:
done